crates/codegen/syntax_templates/src/shared/cst.rs

```diff
@@ -1,167 +1,122 @@
-use std::cmp::{max, min};
-use std::ops::Range;
-use std::rc::Rc;
+use std::{
+    fmt::Display,
+    ops::{Add, AddAssign},
+    rc::Rc,
+};
 
 use serde::Serialize;
 
-use super::kinds::*;
-use super::language::{TextPosition, TextRange};
+use super::language::{RuleKind, TokenKind};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+pub struct RuleNode {
+    pub kind: RuleKind,
+    pub text_size: TextSize,
```

Comment:
suggestion: Maybe "offset" instead of "size", as it is almost always used by users as offsets (and in ranges).

Reply:
we can then hide/not expose `text_size` at all, and always produce offsets, ranges, starts and ends to the user through our APIs.
`TextSize` type can be `TextOffset` as well, to work with `TextRange<>`